### ISSUE TITLE
Zappa throws AccessDenied exception while attempting to upload zip to S3 while AWS CLI can upload files with same profile

### ISSUE THREAD

**Issue by marekyggdrasil** [Miserlou/Zappa#1972]
## Context

Hi all! To make my first deployment using Zappa I cloned your repository and decided to deploy the app from `example` directory.

When I follow your README.md instructions I get `AccessDenied` exception from `botocore`.

Reason why I think it is a bug is I tested my AWS credentials with this `profile` and `bucket` and it uploads files correctly.

```
$ aws --profile <avs config used in zappa> s3 cp <some file> s3://<s3 bucked used with zappa>
upload: ./setup.py to s3://<s3 bucked used with zappa>/<some file>
```

File gets uploaded correctly. When I login to AWS console and open S3 I can see it there.

I made sure I'm using Python 3.6, please check reproduction steps for details.

## Expected Behavior

After running

```
$ zappa deploy dev_api
```

I expect to see the progress bar indicating zip being uploaded to S3, creating API Gateway routes and deploying it.

## Actual Behavior

Instead, I get the following error

```
Calling deploy for stage dev_api..
Creating zappa-permissions policy on ZappaFlaskRole IAM Role.
Downloading and installing dependencies..
 - sqlite==python3: Using precompiled lambda package
'python3.7'
Packaging project as zip.
Oh no! An error occurred! :(

==============

Traceback (most recent call last):
  File "/Users/marek/.pyenv/versions/3.7.4/envs/zappa-project/lib/python3.7/site-packages/zappa/core.py", line 924, in upload_to_s3
    self.s3_client.head_bucket(Bucket=bucket_name)
  File "/Users/marek/.pyenv/versions/3.7.4/envs/zappa-project/lib/python3.7/site-packages/botocore/client.py", line 272, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/marek/.pyenv/versions/3.7.4/envs/zappa-project/lib/python3.7/site-packages/botocore/client.py", line 576, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (403) when calling the HeadBucket operation: Forbidden

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/Users/marek/.pyenv/versions/3.7.4/envs/zappa-project/lib/python3.7/site-packages/zappa/cli.py", line 2779, in handle
    sys.exit(cli.handle())
  File "/Users/marek/.pyenv/versions/3.7.4/envs/zappa-project/lib/python3.7/site-packages/zappa/cli.py", line 509, in handle
    self.dispatch_command(self.command, stage)
  File "/Users/marek/.pyenv/versions/3.7.4/envs/zappa-project/lib/python3.7/site-packages/zappa/cli.py", line 546, in dispatch_command
    self.deploy(self.vargs['zip'])
  File "/Users/marek/.pyenv/versions/3.7.4/envs/zappa-project/lib/python3.7/site-packages/zappa/cli.py", line 723, in deploy
    self.zip_path, self.s3_bucket_name, disable_progress=self.disable_progress)
  File "/Users/marek/.pyenv/versions/3.7.4/envs/zappa-project/lib/python3.7/site-packages/zappa/core.py", line 936, in upload_to_s3
    CreateBucketConfiguration={'LocationConstraint': self.aws_region},
  File "/Users/marek/.pyenv/versions/3.7.4/envs/zappa-project/lib/python3.7/site-packages/botocore/client.py", line 272, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/Users/marek/.pyenv/versions/3.7.4/envs/zappa-project/lib/python3.7/site-packages/botocore/client.py", line 576, in _make_api_call
    raise error_class(parsed_response, operation_name)
botocore.exceptions.ClientError: An error occurred (AccessDenied) when calling the CreateBucket operation: Access Denied

==============

Need help? Found a bug? Let us know! :D
File bug reports on GitHub here: https://github.com/Miserlou/Zappa
And join our Slack channel here: https://slack.zappa.io
Love!,
 ~ Team Zappa!
```

## Steps to Reproduce

1. Login to AWS console and config policy, roles and users as indicated under [this tutorial](https://www.codingforentrepreneurs.com/blog/aws-iam-user-role-policies-zappa-serverless-python), keep bucket name, secret key, arn etc...
2. `aws configure --profile zappa` creates profile name `zappa`, provide credentials from step (1)
2. `git clone git@github.com:Miserlou/Zappa.git --depth 1`, clone repo to fetch the example
3. `cd example` change directory to example
4. Update the `zappa_settings.json` to match one mentioned below in `Your environment` section, mainly to set the local AWS credentials profile, arn etc.
5. `pyenv install 3.6.9` make sure you use `Python 3.6` as recommended
6. `pyenv virtualenv 3.6.9 zappa-project` creates a virtual environment
7. `pyenv local zappa-project` activates the virtual environment
8. `pip install zappa` installs zappa in the virtual environment
9. `zappa deploy dev_api` runs the deployment, here we expect it to fail
10. `aws --profile zappa s3 cp <any-file> s3://<bucket-name>` run this just to upload any file to prove that S3 can be reached with local setup credentials

## Your Environment

Zappa version used:

```
$ zappa -v
0.48.2
```

Operating System and Python version:

```
$ sw_vers
ProductName:    Mac OS X
ProductVersion: 10.14.5
BuildVersion:   18F132
```

* The output of `pip freeze`:

```
$ pip freeze
argcomplete==1.9.3
boto3==1.10.39
botocore==1.13.39
certifi==2019.11.28
cfn-flip==1.2.2
chardet==3.0.4
Click==7.0
docutils==0.15.2
durationpy==0.5
future==0.16.0
hjson==3.0.1
idna==2.8
jmespath==0.9.3
kappa==0.6.0
lambda-packages==0.20.0
placebo==0.9.0
python-dateutil==2.6.1
python-slugify==1.2.4
PyYAML==5.2
requests==2.22.0
s3transfer==0.2.1
six==1.13.0
toml==0.10.0
tqdm==4.19.1
troposphere==2.5.3
Unidecode==1.1.1
urllib3==1.25.7
Werkzeug==0.16.0
wsgi-request-logger==0.4.6
zappa==0.48.2
```

* Your `zappa_settings.json`: 

```json
{
    "dev_api": {
       "project_name": "dev_api_zappa_test_flask_app",
       "keep_warm": false,
       "debug": true,
       "log_level": "DEBUG",
       "aws_region": "<my-region>",
       "s3_bucket": "<my-bucket>",
       "app_function": "app.app",
       "http_methods": ["GET"],
       "parameter_depth": 1,
       "timeout_seconds": 300,
       "memory_size": 128,
       "use_precompiled_packages": true,
       "profile_name": "<my-profile>",
       "role_name": "<my-role>",
       "role_arn": "<my-arn>"
    }
}
```

Where I replace `<my-region>` with the region of my AWS deployment, `<my-bucket>` the name of the bucket I created, `<my-profile>` with local AWS cli profile bound to it and `<my-role>` with role I created in AWS IAM and `<my-arn>` with my ARN string.

**Comment by rudyryk** [comment]
Hi @marekyggdrasil, not clear what IAM permissions are present for the user. As I can see from tracelog ` An error occurred (AccessDenied) when calling the CreateBucket operation` - it's ** CreateBucket** permission is missing. Does that bucket already exist? If so then probably ListBucket permissions are missing.

**Comment by rudyryk** [comment]
Here's the example with basic required permissions:
https://github.com/Miserlou/Zappa/blob/master/example/policy/deploy.json

**Comment by marekyggdrasil** [comment]
Thank you for such quick feedback @rudyryk, you are correct, the bucket indeed does already exist (I created it in advance while preparing according to [this tutorial](https://www.codingforentrepreneurs.com/blog/aws-iam-user-role-policies-zappa-serverless-python)). And it also contains the exact policy used.

I will diff the policy you linked with the one I used, check if it is `ListBucket` permission missing as you predicted and hopedully I can update just the policy on my account without re-doing all the steps again.

I'll post updates here, hopefully after policy update it'll start to work and we could close this issue. Thanks!

**Comment by marekyggdrasil** [comment]
@rudyryk you were absolutely right, `ListBucket` permission has been mentioned in my policy but two things were problematic:

1. I forgot to attach this policy to the user role
2. I made a copy paste error, I forgot to mention bucket name, it has just the default placeholder indicating where to put it

I changed those two problems and now it is deploying (or at least uploading), so fingers crossed for it to finish and thanks a lot for all help!
